### ISSUE TITLE
Update Aeneas to remove the marker traits

### ISSUE THREAD

**Issue by sonmarcho** [AeneasVerif/aeneas#347]
Today, we have to call Charon with the option `--hide-marker-traits`, which is quite annoying and error prone.

**Comment by R1kM** [comment]
Without this option, extracting code such as the following fails:

```rust
pub enum CList<T> {
    CCons(T, Box<CList<T>>),
    CNil,
}
```

This is due to the translation of trait instance ids: https://github.com/AeneasVerif/aeneas/blob/6377992c84f1128103a964b2cf064a3e2e6d7159/compiler/SymbolicToPure.ml#L462

The solution would be to add a prepass to remove these, similarly to what is done in Charon.

**Comment by Nadrieril** [comment]
I'd prefer if aeneas called charon by itself

**Comment by sonmarcho** [comment]
Yes that makes sense. I opened an issue: https://github.com/AeneasVerif/aeneas/issues/349

**Comment by sonmarcho** [comment]
As Aeneas now checks the options with which the LLBC was generated, I consider this issue as closed.
